tests/rest/client/test_auth_issuer.py

```diff
@@ -0,0 +1,61 @@
+# Copyright 2023 The Matrix.org Foundation C.I.C.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from http import HTTPStatus
+
+from synapse.rest.client import auth_issuer
+
+from tests.unittest import HomeserverTestCase, override_config
+
+ISSUER = "https://account.example.com/"
+ACCOUNT_MANAGEMENT_URL = "https://account.example.com/myaccount/"
+
+
+class AccountDataTestCase(HomeserverTestCase):
+    servlets = [
+        auth_issuer.register_servlets,
+    ]
+
+    def test_returns_404_when_msc3861_disabled(self) -> None:
+        # Make an unauthenticated request for the discovery info.
+        channel = self.make_request(
+            "GET",
+            "/_matrix/client/unstable/org.matrix.msc2965/auth_issuer",
+        )
+        self.assertEqual(channel.code, HTTPStatus.NOT_FOUND)
+
+    @override_config(
+        {
+            "disable_registration": True,
+            "experimental_features": {
+                "msc3861": {
+                    "enabled": True,
+                    "issuer": ISSUER,
+                    "account_management_url": ACCOUNT_MANAGEMENT_URL,
+                    "client_id": "David Lister",
+                    "client_auth_method": "client_secret_post",
+                    "client_secret": "Who shot Mister Burns?",
+                }
+            },
+        }
+    )
+    def test_returns_discovery_data_when_oidc_enabled(self) -> None:
+        # Make an unauthenticated request for the discovery info.
+        channel = self.make_request(
+            "GET",
+            "/_matrix/client/unstable/org.matrix.msc2965/auth_issuer",
+        )
+        self.assertEqual(channel.code, HTTPStatus.OK)
+        self.assertEqual(
+            channel.json_body, {"issuer": ISSUER, "account": ACCOUNT_MANAGEMENT_URL}
```

Comment:
That does not match the MSC, BUT I think we'll iterate on it anyway, as there is no real reason for Synapse to serve that info ; we'll probably stick it in the OIDC discovery document instead

Reply:
I hadn't realised that the content of this response differed from the response we serve over the client well-known. I can fix this up to look more like the MSC.

Reply:
I'm going to change the MSC to remove the account URL and action from that endpoint instead to avoid having 3 implementations of this in the wild

Reply:
3074249 should match the latest MSC text.

Reply:
This indeed matches the MSC